From 79128d5fa4456b6ed7b12f622596c4bbd5b28b43 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 14:58:30 -0400
Subject: [infiniband] usnic: backport to older netdev notifier arguments

Message-id: <f596db8068a6e2538251878adcfd5ac432fb6daa.1411569661.git.dledford@redhat.com>
Patchwork-id: 96594
O-Subject: [Patch RHEL7 28/29] usnic: backport to older netdev notifier arguments
Bugzilla: 916382
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

The rhel7 kernels don't include the updated netdevice notifier
that includes the notifier pointer helper function for getting a netdev
out of the opaque data pointer.  Fall back to treating the void *data as
a plain netdev.

Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/usnic/usnic_ib_main.c b/drivers/infiniband/hw/usnic/usnic_ib_main.c
index fb6d026..d35cc86 100644
--- a/drivers/infiniband/hw/usnic/usnic_ib_main.c
+++ b/drivers/infiniband/hw/usnic/usnic_ib_main.c
@@ -214,11 +214,11 @@ static void usnic_ib_handle_usdev_event(struct usnic_ib_dev *us_ibdev,
 }
 
 static int usnic_ib_netdevice_event(struct notifier_block *notifier,
-     unsigned long event, void *ptr)
+     unsigned long event,
+     void *data)
 {
  struct usnic_ib_dev *us_ibdev;
-
- struct net_device *netdev = netdev_notifier_info_to_dev(ptr);
+ struct net_device *netdev = data;
 
  mutex_lock(&usnic_ib_ibdev_list_lock);
  list_for_each_entry(us_ibdev, &usnic_ib_ibdev_list, ib_dev_link) {
-- 
1.7.1